core/src/ledger/storage_api/mod.rs

```diff
@@ -106,24 +118,65 @@ pub trait StorageRead {
 /// Common storage write interface
 pub trait StorageWrite {
     /// Write a value to be encoded with Borsh at the given key to storage.
+    /// Additionally, write the data to the diffs and add it to the merkle tree.
     fn write<T: BorshSerialize>(
         &mut self,
         key: &storage::Key,
         val: T,
     ) -> Result<()> {
         let bytes = val.serialize_to_vec();
-        self.write_bytes(key, bytes)
+        self.write_bytes(key, bytes, WriteActions::All)
+    }
+
```

Comment:
how about we keep `write_bytes` as it (defaulting to `WriteActions::All` and instead only add `write_bytes_with_opts` and `write_with_opts`?

Reply:
done in bd9265b375722fdb25486849c639690c81c64a3a